### PR TITLE
[TEST]Data Service client tests

### PR DESCRIPTION
**Pull request by hudak**
[Build Status](https://travis-ci.org/hudak/pdi-dataservice-plugin)

@lucboudreau

### CODE REVIEW

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. mvn -f pdi-dataservice-client/pom.xml clean install && mvn -f pdi-dataservice-client/pom.xml site

Cleanup:
-------
mvn build-helper:remove-project-artifact

Changed files:
==============
  pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPlugin.java
  pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/client/DataServiceLifecycleListener.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPluginTest.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceLifecycleListenerTest.java

Checkstyle violations:
======================
No errors found

License header violations:
==========================
1. pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPluginTest.java
2. pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceLifecycleListenerTest.java







Unit test coverage change:
============
org.pentaho.di.trans.dataservice.client.DataServiceLifecycleListener:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :90.91%
 * methodPercentCoverage :85.71%
 * complexityPercentCoverage :87.50%
 * instructionPercentCoverage :89.09%
    
org.pentaho.di.trans.dataservice.client.DataServiceClientPlugin:
 * linePercentCoverage :100.00%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :100.00%
 * instructionPercentCoverage :100.00%

**Comment by hudak** [comment]
Added license headers

**Comment by lucboudreau** [comment]
# Build Commands:

## Build:
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. mvn -f pdi-dataservice-client/pom.xml clean install && mvn -f pdi-dataservice-client/pom.xml site

## Cleanup:

mvn build-helper:remove-project-artifact

# Changed files:

  pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPlugin.java
  pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/client/DataServiceLifecycleListener.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPluginTest.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceLifecycleListenerTest.java

# Checkstyle violations:

No errors found

# License header violations:

No violations found

# Unit test coverage change:

org.pentaho.di.trans.dataservice.client.DataServiceLifecycleListener:
- branchPercentCoverage :100.00%
- linePercentCoverage :90.91%
- methodPercentCoverage :85.71%
- complexityPercentCoverage :87.50%
- instructionPercentCoverage :89.09%

org.pentaho.di.trans.dataservice.client.DataServiceClientPlugin:
- linePercentCoverage :100.00%
- methodPercentCoverage :100.00%
- complexityPercentCoverage :100.00%
- instructionPercentCoverage :100.00%

File: pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPlugin.java [hunk 1]
```diff
@@ -155,7 +175,31 @@ public boolean supportsViews() {
...
+  @Override public boolean supportsGetBlob() {
+    return false;
+  }
+
+  @Override public String getCreateTableStatement() {
+    return "// Unsupported";
+  }
+
+  @Override public String getTruncateTableStatement( String tableName ) {
+    return "// Unsupported";
```
Comment (lucboudreau) [inline pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPlugin.java:202]: Your tests do not cover these additions. What's the reason for these?

File: pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      }
+    }
+
+    assertThat( clientPlugin.isFetchSizeSupported(), is( false ) );
+  }
+
+  @Test
+  public void testDDL() throws Exception {
+    assertThat( clientPlugin.getCreateTableStatement(), is( "// Unsupported" ) );
+    assertThat( clientPlugin.getTruncateTableStatement( "dataService" ), is( "// Unsupported" ) );
```
Comment (hudak) [inline pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPluginTest.java:87]: create/Truncate table is tested here

Comment (lucboudreau) [inline pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPluginTest.java:87]: I've added my comment at the end of the list, but there are some in it that are not tested, like supportsEmptyTransactions.

UPDATE: I see. OK.

File: pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  @Test
+  public void testSupportChecks() throws Exception {
+    ImmutableSet<String> supported = ImmutableSet.of(
+      "supportsBooleanDataType", "supportsOptionsInURL", "supportsErrorHandling", "supportsSetMaxRows",
+      "supportsSetLong", "supportsTimeStampToDateConversion"
+    );
+
+    for ( Method method : DatabaseInterface.class.getMethods() ) {
+      String methodName = method.getName();
+      if ( methodName.startsWith( "supports" ) && method.getReturnType() == Boolean.TYPE ) {
```
Comment (hudak) [inline pdi-dataservice-client/src/test/java/org/pentaho/di/trans/dataservice/client/DataServiceClientPluginTest.java:76]: All `boolean supports*()`  methods on DatabaseInterface are tested here.
All are expected to return false unless explicitly named above

**Pull request merged**